projects/plugins/social/src/class-meta-tags.php

```diff
@@ -169,34 +170,128 @@ public function get_featured_image( $post_id = 0, $width = 200, $height = 200 )
+	 *
+	 * @return string $description Cleaned up description string.
+	 */
+	public function get_description( $description = '', $data = null ) {
+		// Remove tags such as <style or <script.
+		$description = wp_strip_all_tags( $description );
+
+		/*
+		 * Clean up any plain text entities left into formatted entities.
+		 * Intentionally not using a filter to prevent pollution.
+		 * @see https://github.com/Automattic/jetpack/pull/2899#issuecomment-151957382
+		 */
+		$description = wp_kses(
+			trim(
+				convert_chars(
+					wptexturize( $description )
+				)
+			),
+			array()
+		);
+
+		// Remove shortcodes.
+		$description = strip_shortcodes( $description );
+
+		// Remove links.
+		$description = preg_replace(
+			'@https?://[\S]+@',
+			'',
+			$description
+		);
+
+		/*
+		 * Limit things to a small text blurb.
+		 * There isn't a hard limit set by Facebook, so let's rely on WP's own limit.
+		 * (55 words or the localized equivalent).
+		 * This limit can be customized with the wp_trim_words filter.
+		 */
+		$description = wp_trim_words( $description );
+
+		// Let's set a default if we have no text by now.
+		if ( empty( $description ) ) {
+			/** This filter is documented in projects/plugins/jetpack/functions.opengraph.php */
+			$description = apply_filters(
+				'jetpack_open_graph_fallback_description',
+				__( 'Visit the post for more.', 'jetpack-social' ),
+				$data
+			);
+		}
+
+		return $description;
+	}
+
 	/**
 	 * Render meta tags in head.
+	 *
+	 * @param WP_Post|null $post The post to render the tags for.
 	 */
-	public function render_tags() {
+	public function render_tags( $post = null ) {
 		if ( ! $this->should_render_meta_tags() ) {
 			return;
 		}
 
-		$image = $this->get_featured_image();
+		$data = get_post( $post );
+		$tags = array();
 
-		if ( empty( $image ) ) {
-			return;
+		if ( empty( $data->post_title ) ) {
+			$tags['og:title'] = __( '(no title)', 'jetpack-social' );
+		} else {
+			/** This filter is documented in core/src/wp-includes/post-template.php */
+			$tags['og:title'] = wp_kses( apply_filters( 'the_title', $data->post_title, $data->ID ), array() );
 		}
 
-		$tags = array(
-			'og:image'        => $image['url'],
-			'og:image:width'  => $image['width'],
-			'og:image:height' => $image['height'],
-		);
+		$tags['og:url'] = get_permalink( $data->ID );
+		if ( ! post_password_required() ) {
+			/*
+			 * If the post author set an excerpt, use that.
+			 * Otherwise, pick the post content that comes before the More tag if there is one.
+			 * Do not use the post content if it contains premium content.
+			 */
+			if ( ! empty( $data->post_excerpt ) ) {
+				$tags['og:description'] = $this->get_description( $data->post_excerpt );
+			} elseif ( ! has_block( 'premium-content/container', $data->post_content ) ) {
+				$excerpt                = explode( '<!--more-->', $data->post_content )[0];
+				$tags['og:description'] = $this->get_description( $excerpt );
+			}
```

Comment:
Not for now, but when we extract this to a package we should make this filterable, so we don't need to check for specific blocks in the package.

Reply:
Good plan. I think there's another filter later, that could be used for this. Definitely something we should do for the package.